src/web/app/pages-instructor/instructor-course-details-page/instructor-course-details-page.component.ts

```diff
@@ -23,27 +33,18 @@ interface CourseStats {
 }
 
 interface CourseDetailsBundle {
-  course: CourseAttributes;
-  stats: CourseStats;
+  course?: CourseAttributes;
```

Comment:
Is it necessary to add the optional mark `?`?

Reply:
I suppose it's not necessary, just have to be more verbose i.e. initialize both `course` and `stats` to have empty properties. Would it be better that way?

Reply:
It is not just for convenience. I consider CourseDetailsBundle as a model in the page. The model should have property `course` and `stats`. That is to say, if course exist, then there must be a stats associated with it. If you put ? before stats, I should read it as `stats for course is optional`

Reply:
That makes sense. Fixed